build_with_distutils_stubs.py

```diff
@@ -0,0 +1,74 @@
+"""Generate distutils stub files inside the source directory before packaging.
+We have to do this as a custom build backend for PEP 660 editable installs.
+Doing it this way also allows us to point local type-checkers to types/distutils,
+overriding the stdlib types even on Python < 3.12."""
```

Comment:
Looks like I got this working without leaving an unchecked `types/` folder in the source.
```suggestion
"""
```